### PR TITLE
Liqun/doc retriever role

### PR DESCRIPTION
**Pull request by liqul**
Re-implement document retriever as a role from the previous plugin.

### CODE REVIEW

File: scripts/document_indexer.py
Comment (ShilinHe) [inline scripts/document_indexer.py:1]: How about moving this to document retriever folder?

File: taskweaver/ext_role/document_retriever/document_retriever.py [hunk 1]
```diff
@@ -2,29 +2,60 @@
...
+        self.vectorstore = None
+        self.embeddings = None
+
+    def initialize(self):
+        import tiktoken
+        from langchain_community.embeddings import HuggingFaceEmbeddings
+        from langchain_community.vectorstores import FAISS
 
         self.embeddings = HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2")
         self.vectorstore = FAISS.load_local(
```
Comment (ShilinHe) [inline taskweaver/ext_role/document_retriever/document_retriever.py:52]: add a check if the local indexing file does not exist? remind the users to run the document_indexer.py file?

Comment (liqul) [reply taskweaver/ext_role/document_retriever/document_retriever.py:52]: The user needs to configure the path explicitly. So, she needs to make sure the existence already.

File: scripts/document_indexer.py
Comment (liqul) [reply scripts/document_indexer.py:1]: All scripts run offline are kept here. I think it is consistent.

**Review by ShilinHe — APPROVED** [review]

**Pull request merged**